proposals/infra/INF-0004-validator-hashing.md

```diff
@@ -127,38 +128,130 @@ apply the hash. Instead it should apply the `PREVIEW` sentinel value from the
 table above. This allows us to differentiate between shaders validated by a
 "final" validator.
 
-### Changes for D3D
+### IDxcValidator API Update
 
-A future AgilitySDK will adopt runtime changes to how and when the shader hash
-is validated and add support for DXIL validation in the debug layer and runtime.
+A new `IDxcValidator` API flag is needed to avoid container hashing and
+unnecessary overhead when used by runtime validation.
+
+Proposed change to flags in `dxcapi.h`:
+
+```cpp
+ static const UINT32 DxcValidatorFlags_Default = 0;
+ static const UINT32 DxcValidatorFlags_InPlaceEdit =
+    1; // Validator is allowed to update shader blob in-place.
+ static const UINT32 DxcValidatorFlags_RootSignatureOnly = 2;
+ static const UINT32 DxcValidatorFlags_ModuleOnly = 4;
++static const UINT32 DxcValidatorFlags_SkipHash =
++    8; // Skip hash update; no modification or copy of input blob.
+ static const UINT32 DxcValidatorFlags_ValidMask = 0xF;
+```
+
+The runtime attempts to use the `DxcValidatorFlags_ValidMask` flag,
```

Comment:
This should have been the `DxcValidatorFlags_SkipHash` flag, which is the newly added one.

```suggestion
The runtime attempts to use the `DxcValidatorFlags_SkipHash` flag,
```